lib/recipe.rb

```diff
@@ -0,0 +1,33 @@
+class Recipe
+    attr_reader :name,
+                :ingredients_required
+
+    def initialize(name)
+        @name                 = name
+        @ingredients_required = {}
+    end
+
+    def add_ingredient(ingredient, quantity)
+        if @ingredients_required[ingredient]
+            @ingredients_required[ingredient] += quantity
+        else 
+            @ingredients_required[ingredient] = quantity
+        end
```

Comment:
Awesome job. Putting a default value on your hash would get rid of the if statement here.